### DISCUSSION TITLE
Best Practice for Simulating Real-World Data with Little Parameter Variations in PyBaMM?

### DISCUSSION THREAD

**Discussion by lappemic** [pybamm-team/PyBaMM#3965] [Q&A] [answered]
Hi everyone,

I'm diving into using PyBaMM for battery simulation and aiming to mirror 'real-world' conditions as closely as possible. Specifically, I want to explore the impact of small variations in battery parameters (such as minimum, average, and maximum voltage) on the overall performance and degradation patterns.

Does anyone have experience or examples of simulating scenarios with these kinds of parameter variations in PyBaMM? Any guidance on best practices or tips for setting up these simulations would be greatly appreciated. I'm looking for insights on how to accurately model and reflect minor differences in cell parameters that often occur in practical applications.

Thanks in advance for your help!

**Comment by TomTranter** [comment]
The voltage is more of an output to the physics-based PyBaMM models not a parameter. Parameters that would cause a change in voltage that might be good to start with are the amount of active material, porosity, particle radius, thickness of electrodes. Then there's a bunch more parameters for degradation. Temperature might affect this the most and rate of cycling more than any physical parameter - trying to fast charge an energy cell with low porosity is going to plate more. Combinations are endless. This might all be less relevant if what your interested in is current and thermal in a pack given variation in cell voltage / soc etc. Need to have a bit more context to offer practical advice

**Comment by lappemic** [comment]
Hey Tom, thanks a lot for your fast and informative reply! 

Sorry for the misunderstanding, the voltage is indeed the output value we are looking for. We want to simulate a batterypack with cells in different health state. Our idea is to run several simulations with PyBaMM and combine their outputs to imitate one batterypack. We want to do this for typical LFP and NMC cells. 

1. Can you recommend a specific battery models for this purpose (LFP and NMC)? 
2. And which parameters would you tune to achieve a natural spread in degradation between the cells? 

Thank you for your support!

**Reply by TomTranter** [reply, marked as answer]
You could check out liionpack in that case. There is an example of running an SEI model in a pack. https://liionpack.readthedocs.io/en/latest/examples/08%20SEI%20degradation%20model/
Probably best thing to do to get some quick results is to play around in PyBaMM with a single cell change the active material and lithium inventory via the initial concentrations. You can use the ElectrodeSoH model to evaluate capacity for a parameter set. Make some of these parameters inputs. Have a read through our examples for setting inputs
